src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -838,6 +838,15 @@ MultiTaskRouterSelectQuerySupported(Query *query)
 
 		Assert(subquery->commandType == CMD_SELECT);
 
+		/* pushing down rtes without relations yields (shardCount * expectedRows) */
+		if (subquery->rtable == NIL)
+		{
+			return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
```

Comment:
Is this a fix for INSERT ... SELECT queries? If yes, shall we do that separately?